testsuite/shared/src/main/java/org/jboss/as/test/integration/domain/management/util/DomainLifecycleUtil.java

```diff
@@ -381,6 +381,54 @@ public void close() {
         stop();
     }
 
+    public void reload(String host) throws IOException, InterruptedException, TimeoutException {
+        reload(host, null, null, null);
+    }
+
+    public void reload(String host, Long timeout) throws IOException, InterruptedException, TimeoutException {
+        reload(host, null, null, timeout);
+    }
+
+    public void reload(String host, Boolean adminOnly, Boolean restartServers) throws IOException, InterruptedException, TimeoutException {
+        reload(host, adminOnly, restartServers, null);
+    }
+
+    /**
+     * Executes a {@code reload} operation and waits a configurable maximum time for the reload to complete.
+     *
+     * @param host the host to use for the request. Cannot be {@code null}
+     * @param adminOnly if {@code true}, the Host Controller will be reloaded in admin-only mode
+     * @param restartServers if {@code true}, servers will be restarted
+     * @param timeout maximum time to wait for the Host Controller and Servers reload to complete, in milliseconds
+     *
+     */
+    public void reload(String host, Boolean adminOnly, Boolean restartServers, Long timeout) throws IOException, InterruptedException, TimeoutException {
+        ModelNode op = new ModelNode();
+        op.get(ModelDescriptionConstants.ADDRESS).add(ModelDescriptionConstants.HOST, host);
+        op.get(ModelDescriptionConstants.OP).set("reload");
+        if (adminOnly != null) {
+            op.get("admin-only").set(adminOnly);
+        }
+        if (restartServers != null) {
+            op.get(ModelDescriptionConstants.RESTART_SERVERS).set(restartServers);
+        }
+        long startupTimeout;
+        if (timeout == null) {
+            startupTimeout = configuration.getStartupTimeoutInSeconds();
+        } else {
+            startupTimeout = timeout;
+        }
+
+        executeAwaitConnectionClosed(op);
+        // Try to reconnect to the hc
+        connect();
+        awaitHostController(System.currentTimeMillis(), startupTimeout);
+        // check that the servers are up
+        if (adminOnly == null || !adminOnly) {
+            awaitServers(System.currentTimeMillis(), startupTimeout);
+        }
```

Comment:
One additional suggestion, even if you are not in admin-only mode, is not always required to wait for your servers.

Could you also make this configurable by using a method argument? If not specified, we can assume it is true.